### DISCUSSION TITLE
How to inherit from the typing without inheriting the "any"?

### DISCUSSION THREAD

**Discussion by paul23-git** [clientIO/joint#2792] [Q&A]
### Introduction

I am in the process of writing a (small) library to provide a more declarative way to describe diagrams, for react like environments.

Instead of pumping everything into a useEffect manually the goals is to have something like:

```
<Paper width={300} height={300} background={{ color: '#F5F5B5' }} cellViewNamespace={namespace}>
    <Rect x={100} y={150} width={80} height={40} callbackState={rect1}/>
    <Rect x={100} y={210} width={80} height={40} callbackState={rect2}/>
    <Link source={rect1[0]} target={rect2[0]} />
</Paper>
```

Now in that proces I was trying to specify the props for Link/Rect. Not wishing to completely reinvent the wheel myself, I Tried to make due with the types already supplied by jointjs:  `shapes.standard.LinkAttributes`

While this works, I also notice that all my typing stopped giving warnings when I used or gave unsupported attributes. Anything can be supplied and is then defined as "any".

I noticed that this is due to the typing from `dia.Cell.GenericAttributes` which has:

        interface GenericAttributes<T> {
            attrs?: T;
            z?: number;
            [key: string]: any;
        }

Why is that is that line in there? Can we remove that when I inherit from that?

### Steps to reproduce

a question...

### Restrictions & Constraints

a question...

### Does your question relate to JointJS or JointJS+. Select both if applicable.

JointJS

**Comment by kumilingus** [comment]
Hi!

> Why is that is that line in there?

SVG includes visual attributes like `fill` and `stroke`. However, since it's a DOM node, you can also add custom attributes, such as `data-myattribute="1"` and so on  while keeping the SVG document valid.

> Can we remove that when I inherit from that?

I guess it's ok to remove it for your use-case. For instance, using https://stackoverflow.com/a/58223140/2921495.
